adapters/smartadserver/smartadserver.go

```diff
@@ -0,0 +1,186 @@
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "Error parsing bidderExt object",
+			})
+			continue
+		}
+
+		var smartadserverExt openrtb_ext.ExtImpSmartadserver
+		if err := json.Unmarshal(bidderExt.Bidder, &smartadserverExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "Error parsing smartadserverExt parameters",
+			})
+			continue
+		}
+
+		// Adding publisher id.
+		smartRequest.Site.Publisher.ID = strconv.Itoa(smartadserverExt.NetworkID)
+
+		// We send one request for each impression.
+		var monoImp []openrtb.Imp
+		monoImp = append(monoImp, imp)
+		smartRequest.Imp = monoImp
+
+		var errMarshal error
+		if imp.Ext, errMarshal = json.Marshal(smartadserverExt); errMarshal != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: errMarshal.Error(),
+			})
+			continue
+		}
+
+		reqJSON, err := json.Marshal(smartRequest)
+		if err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "Error parsing reqJSON object",
+			})
+			continue
+		}
+
+		url, err := a.BuildEndpointURL(&smartadserverExt)
+		if url == "" {
+			errs = append(errs, err)
+			continue
+		}
+
+		headers := http.Header{}
+		headers.Add("Content-Type", "application/json;charset=utf-8")
+		headers.Add("Accept", "application/json")
+		adapterRequests = append(adapterRequests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     url,
+			Body:    reqJSON,
+			Headers: headers,
+		})
+	}
+	return adapterRequests, errs
+}
+
+// MakeBids unpacks the server's response into Bids.
+func (a *SmartadserverAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{fmt.Errorf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode)}
+	}
+
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	for _, sb := range bidResp.SeatBid {
+		for i := 0; i < len(sb.Bid); i++ {
+			bid := sb.Bid[i]
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: getMediaTypeForImp(bid.ImpID, internalRequest.Imp),
+			})
+
+		}
+	}
+	return bidResponse, []error{}
+}
+
+// BuildEndpointURL : Builds endpoint url
+func (a *SmartadserverAdapter) BuildEndpointURL(params *openrtb_ext.ExtImpSmartadserver) (string, error) {
+
+	host := a.host
+	if params.Domain != "" {
```

Comment:
Why would a bidder need to define their own domain? Is it entirely free form?

Reply:
We used this domain parameter as a convenient way to call our local&development testing environments. 
Indeed, it's not really for bidders.

Reply:
Ah, ok. Thank you for the context. Could you please remove this ability? It would enable the adapter to act as a publisher defined proxy, which we'd like to avoid. For testing, please use a config or environment variable override to specify local host (that's what we do).